src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -2960,35 +2960,34 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
         //
         // Note that method values are a separate thing (`m _`): they have the idiosyncratic shape
         // of `Typed(expr, Function(Nil, EmptyTree))`
-        val ptUnrollingEtaExpansion =
-          if (paramsMissingType.nonEmpty && pt != ErrorType) fun.body match {
+        val (fun2, ptUnrollingEtaExpansion) =
+          if (paramsMissingType.isEmpty || pt == ErrorType) (fun, null)
+          else fun.body match {
             // we can compare arguments and parameters by name because there cannot be a binder between
             // the function's valdefs and the Apply's arguments
             case Apply(meth, args) if (vparams corresponds args) { case (p, Ident(name)) => p.name == name case _ => false } =>
               // We're looking for a method (as indicated by FUNmode in the silent typed below),
               // so let's make sure our expected type is a MethodType
               val methArgs = NoSymbol.newSyntheticValueParams(argpts map { case NoType => WildcardType case tp => tp })
-
-              val result = silent(_.typed(meth, mode.forFunMode, MethodType(methArgs, respt)))
```

Comment:
Let's explore keeping the `silent` and a variation of throwing in a `.duplicate`  -> `silent(_.typed(meth.duplicate, mode.forFunMode, MethodType(methArgs, respt)))`. That fixed the issue for me as well. I'm hesitant to drop the `silent`, but grant that duplication is costly.

Reply:
IIRC there was a previous PR that duplicates? For some reason, it seemed simpler just to accept the unsilent judgment; but I don't spend all day with the code, and also I don't remember what I did last summer. Maybe this is an opportunity to try some jmh-driven development.